File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        profit = 0
        l = 0
        for r in range(1,len(prices)):
            if prices[l] < prices[r]:
                profit = max(profit, prices[r]-prices[l])
            else:
                l = r
        return profit
        

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int profit = 0;
        int mini = Integer.MAX_VALUE;
        for(int i=0;i<prices.length;i++){
            mini = Math.min(mini, prices[i]);
            profit = Math.max(profit, prices[i]-mini);
        }
        return profit;
    }
}